### PR TITLE
Add binary validation

### PR DESCRIPTION
**Pull request by egorikftp**

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -27,6 +27,7 @@ kotlin-test = { module = "org.jetbrains.kotlin:kotlin-test", version.ref = "kotl
 [plugins]
 android-application = { id = "com.android.application", version.ref = "agp" }
 android-library = { id = "com.android.library", version.ref = "agp" }
+binary-compatibility = { id = "org.jetbrains.kotlinx.binary-compatibility-validator", version = "0.16.2" }
```
Comment (Goooler) [inline gradle/libs.versions.toml:30]:
```suggestion
binary-compatibility = "org.jetbrains.kotlinx.binary-compatibility-validator:0.16.2"
```

**Comment by Goooler** [comment]
Can also enable https://github.com/Kotlin/KEEP/blob/master/proposals/explicit-api-mode.md

**Comment by egorikftp** [comment]
> Can also enable https://github.com/Kotlin/KEEP/blob/master/proposals/explicit-api-mode.md

@vkatz what do you think?

**Comment by vkatz** [comment]
@egorikftp we can try to play around ` explicitApi()` for lib modules

**Comment by egorikftp** [comment]
will cover in separate ticket

**Review by vkatz — APPROVED** [review]

**Pull request merged**